src/main/java/com/mraof/minestuck/util/MSAttachments.java

```diff
@@ -0,0 +1,87 @@
+			() -> AttachmentType.builder(restricted(() -> ColorHandler.BuiltinColors.DEFAULT_COLOR, PlayerData.class)).serialize(Codec.INT).build());
+	public static final Supplier<AttachmentType<Long>> BOONDOLLARS = REGISTER.register("boondollars",
+			() -> AttachmentType.builder(restricted(() -> 0L, PlayerData.class)).serialize(Codec.LONG).build());
+	public static final Supplier<AttachmentType<GristCache>> GRIST_CACHE = REGISTER.register("grist_cache",
+			() -> AttachmentType.serializable(restricted(GristCache::new, PlayerData.class)).build());
+	public static final Supplier<AttachmentType<Double>> GUTTER_MULTIPLIER = REGISTER.register("gutter_multiplier",
+			() -> AttachmentType.builder(restricted(() -> 1D, PlayerData.class)).serialize(Codec.DOUBLE).build());
+	public static final Supplier<AttachmentType<Echeladder>> ECHELADDER = REGISTER.register("echeladder",
+			() -> AttachmentType.serializable(restricted(Echeladder::new, PlayerData.class)).build());
+	
+	public static final Supplier<AttachmentType<Title>> TITLE = REGISTER.register("title",
+			() -> AttachmentType.<Title>builder(noDefault()).serialize(Title.CODEC).build());
+	public static final Supplier<AttachmentType<Boolean>> EFFECT_TOGGLE = REGISTER.register("effect_toggle",
+			() -> AttachmentType.builder(restricted(() -> false, ServerPlayer.class)).serialize(Codec.BOOL).build());
+	public static final Supplier<AttachmentType<EditmodeLocations>> EDITMODE_LOCATIONS = REGISTER.register("editmode_locations",
+			() -> AttachmentType.serializable(EditmodeLocations::new).build());
+	public static final Supplier<AttachmentType<ConsortReputation>> CONSORT_REPUTATION = REGISTER.register("consort_reputation",
+			() -> AttachmentType.serializable(restricted(ConsortReputation::new, PlayerData.class)).build());
+	
+	public static final Supplier<AttachmentType<ItemStackHandler>> MUSIC_PLAYER_INVENTORY = REGISTER.register("music_player_inventory",
+			() -> AttachmentType.serializable(restricted(() -> new ItemStackHandler(1), ItemStack.class)).build());
+	public static final Supplier<AttachmentType<MusicPlaying>> MUSIC_PLAYING = REGISTER.register("music_playing",
+			() -> AttachmentType.builder(restricted(MusicPlaying::new, LivingEntity.class)).build());
+	
+	public static final Supplier<AttachmentType<EditTools>> EDIT_TOOLS = REGISTER.register("edit_tools",
+			() -> AttachmentType.builder(restricted(EditTools::new, Player.class)).build());
+	public static final Supplier<AttachmentType<MSFluidType.LastFluidTickData>> LAST_FLUID_TICK = REGISTER.register("last_fluid_tick",
+			() -> AttachmentType.builder(restricted(MSFluidType.LastFluidTickData::new, LivingEntity.class)).build());
+	public static final Supplier<AttachmentType<DialogueComponent.CurrentDialogue>> CURRENT_DIALOGUE = REGISTER.register("current_dialogue",
+			() -> AttachmentType.builder(restricted(DialogueComponent.CurrentDialogue::new, ServerPlayer.class)).build());
```

Comment:
Comment on layout, I think it would make sense to group the following together
- edit tools and editmode locations
- current dialogue and consort reputation